qiskit_research/utils/pauli_twirling.py

```diff
@@ -109,19 +111,51 @@ def run(
     ) -> DAGCircuit:
         for run in dag.collect_runs(list(self.gates_to_twirl)):
             for node in run:
-                twirl_gates = TWIRL_GATES[node.op.name]
-                (before0, before1), (after0, after1) = twirl_gates[
-                    self.rng.integers(len(twirl_gates))
-                ]
-                mini_dag = DAGCircuit()
-                register = QuantumRegister(2)
-                mini_dag.add_qreg(register)
-                mini_dag.apply_operation_back(before0, [register[0]])
-                mini_dag.apply_operation_back(before1, [register[1]])
-                mini_dag.apply_operation_back(node.op, [register[0], register[1]])
-                mini_dag.apply_operation_back(after0, [register[0]])
-                mini_dag.apply_operation_back(after1, [register[1]])
-                dag.substitute_node_with_dag(node, mini_dag)
+                if len(node.op.params) == 1:  # supersede the parameterized list
+                    mini_dag = DAGCircuit()
+                    q0, q1 = node.qargs
+                    mini_dag.add_qreg(q0.register)
+
+                    theta = node.op.params[0]
+                    this_pauli = Pauli(
+                        self.rng.choice(pauli_basis(2).to_labels())
+                    ).to_instruction()
+                    if node.op.name[0] == "r":
+                        if Pauli(node.op.name.split("r")[1].upper()[::-1]).anticommutes(
```

Comment:
Depending on the name of the gate beginning with "r" is precarious. For example, any user-define gate starting with "r" would break this, right? Instead, I think we should explicitly handle a fixed list of supported gates.

Reply:
That is correct, the logic now checks that the `Gate` (`node.op` here) is indeed a 2-qubit rotation gate. However, I have not found a better way to extract the `Pauli` information from that gate other than string manipulation.